clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -296,9 +281,11 @@ public boolean awaitPendingRequests(Node node, long timeoutMs) {
      */
     public int pendingRequestCount(Node node) {
         synchronized (this) {
-            List<ClientRequest> pending = unsent.get(node);
-            int unsentCount = pending == null ? 0 : pending.size();
-            return unsentCount + client.inFlightRequestCount(node.idString());
+            int total = 0;
+            for (UnsentRequest request : unsent)
```

Comment:
@hachikuji Both unit tests and the extra method and it is not difficult at all. But I personally don't get why we bother to do it. If we are worrying about the complexity of this approach then why don't we just use `synchronize(unsent)`, which guarantees better worst-case performance for per-node operation. I adopted your suggestion under the assumption that the extra complexity doesn't matter. Otherwise I would argue that we adopt the original approach.